clang/include/clang/CIR/Dialect/IR/CIROps.td

```diff
@@ -1246,6 +1246,59 @@ def SelectOp : CIR_Op<"select", [Pure,
   }];
 }
 
+//===----------------------------------------------------------------------===//
+// TernaryOp
+//===----------------------------------------------------------------------===//
+
+def TernaryOp : CIR_Op<"ternary",
+      [DeclareOpInterfaceMethods<RegionBranchOpInterface>,
+       RecursivelySpeculatable, AutomaticAllocationScope, NoRegionArguments]> {
+  let summary = "The `cond ? a : b` C/C++ ternary operation";
+  let description = [{
+    The `cir.ternary` operation represents C/C++ ternary, much like a `select`
+    operation. The first argument is a `cir.bool` condition to evaluate, followed
+    by two regions to execute (true or false). This is different from `cir.if`
+    since each region is one block sized and the `cir.yield` closing the block
+    scope should have one argument.
```

Comment:
Do we represent the GNU extension of `x = Thing1 ?: Thing2;` any differently?

Reply:
GNU `?:` is only partially implemented (we lack glvalue support for opaque value expressions) but it's represented by `cir.ternary`.

Reply:
Can we document it here?  One of the 'features' of it IIRC is that it only evaluates `Thing1` 1x instead of 2x (like in a traditional ternary), so it might be nice to document how that is represented here.